test/types/coerce/allNumericsUnaryRealImagComplex.chpl

```diff
@@ -19,11 +19,6 @@ proc f(x: complex(128)) {
   writeln("  f(", x.type:string, ")");
 }
 
-proc f(x) {
-  writeln("  f(generic)");
-}
-
-
```

Comment:
I take it that removing this keeps the test working as-is, but that keeping it in would cause it to be called in all non-real/imag/complex cases?  That could be interesting to capture in a second variation of this test (unless it's already covered elsewhere), since if effectively shows how the behavior of this test has changed (i.e., I read the test more as "which is called in these cases?" rather than "are the coercions happening as expected?".  But I see you're the test's author, so you probably know best.

Reply:
[note: I'm reviewing the tests bottom-to-top, which is why I didn't note this in the previous cases...]

Reply:
Yeah. It's probably reasonable to add a variation along these lines. I'm include to focus on a variation of `allNumericsBinary.chpl`, just because there are a lot of tests here. But we could make the variation for `allNumericsUnaryRealImagComplex.chpl` and `allNumericsBinary.chpl` if you think that has value.

Reply:
That proposal sounds fine to me, though I'll leave the final decision up to you since, again, you're the author of the test.

Reply:
I've added allNumericsBinaryWithGeneric.chpl and allNumericsUnaryRealImagComplexWithGeneric.chpl. allNumericsBinaryWithGeneric.chpl fails to compile with an ambiguity error. It's unclear to me if that is a problem. But, anyway, now we have a test to know if that changes.